CHANGELOG.rst

```diff
@@ -8,6 +8,44 @@ inverse chronological order.
 Changes since version 5.0
 -------------------------
 
+- Fixed and fallback lengths in (namespace) expressions
+
+  Sometimes an expression cannot be parsed because the shape cannot be
+  determined. A typical example is the solitary identity array ``δ_ij``:
+
+      >>> ns = nutils.function.Namespace()
+      >>> ns.eval_ij('δ_ij')
+      Traceback (most recent call last):
+      ...
+      nutils.expression.ExpressionSyntaxError: Length of axis cannot be determined from the expression.
+      δ_ij
+        ^
+
+  The :class:`nutils.function.Namespace` now has two methods to solve this
+  problem. You can specify a fallback length:
+
+      >>> ns = nutils.function.Namespace(fallback_length=2)
+      >>> ns.eval_ij('δ_ij')
+      Array<2,2>
+
+  Or you can fix an index to a length, e.g.
+
+      >>> ns = nutils.function.Namespace(fixed_lengths=dict(i=2))
+      >>> ns.eval_ij('δ_ij')
+      Array<2,2>
+
+  Note that evaluating an expression with an incompatible length raises an
+  exception:
+
+      >>> ns = nutils.function.Namespace(fixed_lengths=dict(i=2))
+      >>> ns.a = numpy.array([1,2,3])
+      >>> 'a_i' @ ns
+      Traceback (most recent call last):
+      ...
+      ExpressionSyntaxError: Length of index i is fixed at 2 but the expression has length 3.
+      a_i
+        ^
+
```

Comment:
I suggest we keep changelog entries fairly brief and link to the API reference (function.Namespace) for details. In any case the last paragraph (incompatible lengths) should be included in the Namespace docstring.

Reply:
I've reduced the changelog entry.